### PR TITLE
"Share with" button

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
Closes https://github.com/enso-org/cloud-v2/issues/497
- Implements "share with" modal
- Invites the user to the organization instead if the email is not found

### Important Notes
- The design is temporary, and needs to be replaced with a proper design later.
- After adding a permission, the row is not updated. This is intentional; it can(/will?) be fixed in the PR allowing updates to individual rows: #6546

### Screencast

[screen-recorder-thu-jun-22-2023-19-10-07.webm](https://github.com/enso-org/enso/assets/4046547/5165f40f-c8a6-4ac5-8720-6e31b053b958)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by PabloBuchu** [comment]
Can we do something how it looks like? the `+` sign is not centered, options are being below permission actions, there is not padding around email text input, `+` is wandering below the avatars..

Screenshot 2023-06-23 at 15 40 30
Screenshot 2023-06-23 at 15 40 18
Screenshot 2023-06-23 at 15 44 12


When I select a user the share with doesn't not appear until I reload. My email shouldn't be listed

**Comment by PabloBuchu** [comment]
Couple of issues spotted:
1. When I add multiple users the share with is shrinking:
Screenshot 2023-06-26 at 15 11 02

It should go like in figma:
Screenshot 2023-06-26 at 15 11 33

With on hover the avatar should go in the foreground and when click we should have the same modal pre populated with selected user and be able to update permissions (I will add support for that on backend)


2. The permission actions should be multiple select (also need backend to support that, you can send action in a list)
Screenshot 2023-06-26 at 15 11 15

**Comment by wdanilo** [comment]
A few things here:
1. It should be possible to easily share that with multiple people. Having to open the share dialog multiple times is a bad UX, it should be possible to do it easily.
2. Next to every person that I add, there should be a way to assign a person to any of the defined in the Figma states:

image

I still don't know how to do it nicely visually, but the functionality should be there. I will try to mock something soon.

**Comment by PabloBuchu** [comment]
`Read` is now `View` so this is why it throws 400

**Comment by PabloBuchu** [comment]
Minor bug when I open a sharing option for edition

Screenshot 2023-06-30 at 12 53 55

**Comment by PabloBuchu** [comment]
I guess also we should filter out from the email dropdown users the assets was already shared with

**Comment by somebody1234** [comment]
nice catches, both fixed

**Comment by xvcgreg** [comment]
@PabloBuchu what's the holdup here?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts:307]: why name change? could we keep it consistent and use `directory` here also?

**Review by PabloBuchu — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts:307]: I think for all user facing messages we have been using "folder" instead of "directory"... but definitely possible to change if desired

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts:307]: *to clarify: The Figma uses "Folders", so that's what I've been using for any user facing text

**Comment by PabloBuchu** [comment]
@somebody1234 one thing. the `share with` button should be visible only for the `Owners`. Currently I shared an asset with my second account for view and I was able to `Own` it :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts
Comment (PabloBuchu) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts:307]: oh ok then lets leave it :)

**Comment by PabloBuchu** [comment]
@somebody1234 one small improvement. The selected action shadow is too bright can we make it more visible? or when I select permission to edit no action is selected

**Review by indiv0 — APPROVED** [review]

**Comment by PabloBuchu** [comment]
@wdanilo would you be able to do a review here? We need your approval since we are not fully code owners (`/assets` directory).

We still working on multi select for users (so you can share the given asset with multiple people ot once). We improved with selecting multiple permissions actions so you don't need so granulated permissions (like in figma). It feels more descriptive that way.

My idea is to merge this as a first implementation and then do a longer live session working with the whole dashboard (preferably with @sylwiabr @jdunkerley @xvcgreg) and see what could be improved in UX in general. WDYT?

https://github.com/enso-org/enso/assets/2855109/dcb52305-9e6c-4059-aeb5-07efa167292f

**Review by wdanilo — APPROVED** [review]

**Pull request merged**
